### ISSUE TITLE
stalkerhek.yml

### ISSUE THREAD

**Issue by sebagsm** [erkexzcx/stalkerhek#1]
hi!

can you post a little more how to about install and setup? missing stalkerhek.yml. thanks :)

**Comment by kettenbach-it** [comment]
stalkerhek.yml is simple:

```
model: MAG254
serial_number: 123456789012
device_id: XXXXXXXXXXXXXXXXXXXXXXXX95268566D2116116D622EAFFD134BF4A9403485A9A
device_id2: XXXXXXXXXXXXXXXXXXXXXXXXXXXX1F95684B12329FDD542566C52811D134B33DCE1C
signature:
mac: 00:12:34:56:78:90
username:
password:
.....
```
For the config options take a look at this https://github.com/erkexzcx/stalkerhek/blob/master/internal/config/config.go

For the values you need to use tcpdump or wireshark and dump a GET-request of your box to the "portal.php" on you providers stalker portal

**Comment by ashwincr** [comment]
I have this running with the parameters that I snooped from both stbemu and ott navigator both of which work with my portal. I had the server running once, but it since then has errored out. Even when it was running, it generated the m3u file, I could load it up in vlc and see the channels, but the server gave an error when I tried to play it. I haven't been able to get it to authenticate since then and I can't exactly remember what the error was. 
I can help troubleshoot and contribute if I can, anf I think something like this will be awesome for watching IPTV

**Comment by dotur911** [comment]
Yes,excatly as ashwincr said, the channels after generated list and loaded in vlc opens as list, but start skipping and not opening nither one, jusk skipping. Please is there anything to fix this problem. I need this badly to get it working. Thank you in advance to this wonderful project.

**Comment by erkexzcx** [comment]
First of all I've never thought that such random project would get this amount of interest. Anyway, let's focus on this issue:

> the channels after generated list and loaded in vlc opens as list, but start skipping and not opening nither one, jusk skipping.

Actually I very well know what you mean and got this "bug" myself dozens of times. Here are some possible scenarios why this is happening:
- I implemented mechanism incorrectly and it does not update channel when it's actually needed, therefore trying to re-use expired links. Should be [this line](https://github.com/erkexzcx/stalkerhek/blob/master/internal/proxy/request_iptv.go#L23) which leads to [this function](https://github.com/erkexzcx/stalkerhek/blob/af317756556d1454010f8164810d48990759cd4a/internal/proxy/utils.go#L46) and all the logic behind. If more than 30 seconds passed since this channel was last queried ("viewed"), then it attempts to get new URL (with token?) from Stalker portal.
- Could be [this function](https://github.com/erkexzcx/stalkerhek/blob/master/pkg/stalker/channels.go#L22) that is implemented incorrectly and is supposed to build a new URL (aka "get new URL") from Stalker portal. This also includes headers (either missing or unexpected).
- Could be other issues I am not aware of right away.

Long story short - I have close to no information at all about stalker portals and pretty much everything was reverse engineered from IPTV box where credentials where pre-configured. :)

If you want to help me debug it - feel free to spin-up wireshark, record where your STB connects (e.g. channel's URL), then do the same with this app and provide me comparison/differences. Right now I am not sure if I can help you, but if you provide some useful information, then I can take a look and potentially fix this issue.

**Comment by erkexzcx** [comment]
Feel free to share with me Stalker credentials so I can troubleshoot it. Otherwise I have no info on how Stalker works and I am unable to fix any of your issues.

**Comment by erkexzcx** [comment]
With latest commits I think I've fixed this issue. Please check against latest code in this repo and if it still doesn't work - open a new issue.
